nifi-nar-bundles/nifi-azure-bundle/nifi-azure-reporting-task/src/main/java/org/apache/nifi/reporting/azure/loganalytics/api/AzureLogAnalyticsMetricsFactory.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.reporting.azure.loganalytics.api;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.apache.nifi.metrics.jvm.JvmMetrics;
+import org.apache.nifi.processor.DataUnit;
+import org.apache.nifi.controller.status.ConnectionStatus;
+import org.apache.nifi.controller.status.ProcessGroupStatus;
+import org.apache.nifi.controller.status.ProcessorStatus;
+import org.apache.nifi.reporting.azure.loganalytics.MetricNames;
+import org.apache.nifi.reporting.azure.loganalytics.Metric;
+import org.apache.nifi.reporting.azure.loganalytics.MetricsBuilder;
+
+
+public class AzureLogAnalyticsMetricsFactory {
+
+    public static List<Metric> getDataFlowMetrics(ProcessGroupStatus status, String instanceId){
+
+        String groupId = status.getId();
```

Comment:
I know there are always arguments about excessive use final, so I won't emplace my opinion here; however, the method, below uses final for the assignments. Should be consistent, when possible.